src/Workspaces/CoreTest/WorkspaceServiceTests/TemporaryStorageServiceTests.cs

```diff
@@ -323,5 +324,24 @@ public void StreamTest3()
                 }
             }
         }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/353"), Trait(Traits.Feature, Traits.Features.Workspace)]
+        public void TestTemporaryStorageTextEncoding()
+        {
+            var textFactory = new TextFactoryService();
+            var service = new TemporaryStorageServiceFactory.TemporaryStorageService(textFactory);
+
+            // test normal string
+            var text = SourceText.From(new string(' ', 4096) + "public class A {}", Encoding.ASCII);
+            TestTemporaryStorage(service, text);
```

Comment:
This (TestTemporaryStorage) doesn't appear to test the encoding, which was the whole point of adding this test, wasn't it?